reference/src/layout/unions.md

```diff
@@ -6,29 +6,108 @@ not to change until an RFC ratifies them.
 
 [#13]: https://github.com/rust-rfcs/unsafe-code-guidelines/issues/13
 
-The only degree of freedom the compiler has when computing the layout of a union
-like
+### Layout of individual union fields
 
-```rust,ignore
-union U { f1: T1, f2: T2 }
+A union consists of several variants, one for each field. All variants have the
+same size and start at the same memory address, such that in memory the variants
+overlap. This can be visualized as follows:
+
+```text
+[ <--> [field0_ty] <----> ]
+[ <----> [field1_ty] <--> ]
+[ <---> [field2_ty] <---> ]
 ```
 
-is to determine the offset of the fields. The layout of these fields themselves
-is already entirely determined by their types, and since we intend to allow
-creating references to fields (`&u.f1`), unions do not have any wiggle-room
-there.
+**Figure 1** (union-field layout): Each row in the picture shows the layout of
+the union for each of its variants. The `<-...->` and `[ ... ]` denote the
+differently-sized gaps and fields, respectively.
+
+The individual fields (`[field{i}_ty_]`) are blocks of fixed size determined by
+the field's [layout]. The only degrees of freedom the compiler has when
+computing the layout of a union are the size of the union, which can be larger
+than the size of its largest field, and the offset of each union field within
+its variant. How these are picked depends on certain constraints like, for
+example, the alignment requirements of the fields, the `#[repr]` attribute of
+the `union`, etc.
+
+[padding]: ../glossary.md#padding
+[layout]: ../glossary.md#layout
```

Comment:
Shouldn't this be `glossary.html` to work in the rendered book?

Reply:
I don't think so. The links should be to the markdown files, and then whether you generate html, or latex, or something else will depend on the markdown backend used.

Reply:
Does the book compiler fix those links? Certainly in the past we have used HTML link, e.g. [here](https://github.com/rust-lang/unsafe-code-guidelines/blame/2bda4cbb469df96044a2aae47840cb34cee5edc8/reference/src/layout/enums.md#L379).

Reply:
OTOH, links actually do seem to get adjusted by HTML generation. So I agree with you, your links are right -- and the others are wrong.

Reply:
> Does the book compiler fix those links? 

Yes, it does.

> Certainly in the past we have used HTML link, e.g. here.

Damn, we shouldn't do that. I'll fix that in a different PR.

Reply:
I've opened: https://github.com/rust-lang-nursery/mdBook/issues/1019